modules/battery_manager/php/testprovisioner.class.inc

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * This file implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\battery_manager;
+
+/**
+ * This class implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class TestProvisioner extends \LORIS\Data\ProvisionerInstance
+{
+    /**
+     * Select all tests from the test_battery table that match the attribute
+     * values passed as conditions and will return an array of test objects
+     * indexed by id
+     *
+     * @param array $conditions Conditions to be met by the query
+     *
+     * @return array $pools     List of Test Objects that match the query
+     */
+    public function selectInstances(array $conditions = null) : array
+    {
+        $db = \Database::singleton();
+        $query = "SELECT b.id,
+                         t.Test_name as testName,
+                         b.AgeMinDays as ageMinDays,
+                         b.AgeMaxDays as ageMaxDays,
+                         b.Stage as stage,
+                         s.SubprojectID as subproject,
+                         b.Visit_label as visitLabel,
+                         p.CenterID as centerId,
+                         b.firstVisit,
+                         b.instr_order as instrumentOrder,
+                         b.Active as active
+                  FROM test_battery b
+                    LEFT JOIN test_names t USING (Test_name)
+                    LEFT JOIN subproject s USING (SubprojectID)
+                    LEFT JOIN psc p USING (CenterID)
+                  ORDER BY Full_name";
+        $query .= $conditions ? ' WHERE '.$this->db->implodeWithKeys(' AND ', $conditions) : '';
```

Comment:
I don't know what this is supposed to be used for, but it seems like it should be using a filter to the provisioner instead.

But from what I can tell it's never passed anything other than null so it isn't used (which I guess is good, because  it would result in an SQL syntax error if it were..)